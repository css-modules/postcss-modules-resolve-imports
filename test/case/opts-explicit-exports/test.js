'use strict';

const resolveImports = require('../../../index');
const setup = require('../../setup');
const test = require('tape');

test('opts-explicit-exports', t => {
  const {expected, resulting, exports: tokens} = setup(
    'local-by-default',
    'extract-imports',
    'scope',
    resolveImports({icssExports: true})
  )(__dirname);

  t.equal(resulting.trim(), expected.trim());
  t.deepEqual(tokens, {
    continueButton: '_source_continueButton',
    userInput: '_source_userInput',
  });
  t.end();
});
