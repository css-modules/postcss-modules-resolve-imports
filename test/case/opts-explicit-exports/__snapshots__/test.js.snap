// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opts-explicit-exports 1`] = `
":export
{
  continueButton: _source_continueButton;
  userInput: _source_userInput;
}

/* imported from source.css */

._source_continueButton
{
  color: green;
}

._source_userInput
{
  /**/
}
"
`;

exports[`opts-explicit-exports 2`] = `
Object {
  "continueButton": "_source_continueButton",
  "userInput": "_source_userInput",
}
`;
